src/main/java/org/apache/maven/buildcache/CachingMojosExecutionStrategy.java

```diff
@@ -202,7 +202,6 @@ private boolean restoreProject( CacheResult cacheResult,
             // cleanup partial state
             project.getArtifact().setFile( null );
             project.getArtifact().setResolved( false );
-            project.getAttachedArtifacts().clear();
```

Comment:
a hotfix, as the attached artifacts is an unmodifiable list. Perhaps need a better fix.